packages/aws-cdk-lib/aws-elasticloadbalancingv2-targets/lib/alb-target.ts

```diff
@@ -32,8 +36,12 @@ export class AlbArnTarget implements elbv2.INetworkLoadBalancerTarget {
 }
 
 /**
- * A single Application Load Balancer as the target for load balancing.
- */
+  * A single Application Load Balancer as the target for load balancing.
+  * @deprecated Use AlbListenerTarget instead. This target does not automatically
+  * add a dependency between the ALB listener and resulting NLB target group, without
+  * which may cause stack deployments to fail if the NLB target group is provisioned
+  * before the listener has been fully created.
+  */
```

Comment:
An alternative would be to keep using `AlbTarget` and set dependencies using [`ApplicationLoadBalancer.listeners`](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_elasticloadbalancingv2.IApplicationLoadBalancer.html#listeners). Which do you think is better?

Reply:
This is due to the same reason provided on this following PR - https://github.com/aws/aws-cdk/pull/17248#issue-1040439161

> While `AlbTarget` could have logic added to it to add dependencies, this would not work if the provided ALB doesn't have any items in its listener property (e.g. `new ApplicationListener(...)` instead of `alb.addListener(...)`)

Reply:
Thanks, that makes sense.

Reply:
@stm29 My concern was that this change will make imported ALBs not usable as a NLB target. But I found that they can still use `AlbArnTarget` for that purpose.  How about adding a comment like: `Use AlbListenerTarget instead or AlbArnTarget for an imported load balancer.`

Reply:
@tmokmss , 
>My concern was that this change will make imported ALBs not usable as a NLB target

Got it

Modified comment as follows:
```
Use `AlbListenerTarget` instead or `AlbArnTarget` for an imported load balancer
```